src/coreComponents/finiteVolume/ProjectionEDFMHelper.hpp

```diff
@@ -0,0 +1,107 @@
+#ifndef __PROJECTIONEDFMHELPER_H_
+#define __PROJECTIONEDFMHELPER_H_
+
+#include "finiteVolume/FluxApproximationBase.hpp"  // FIXME: include appropriate explicitly
+#include <vector>                                  // provides std::vector
+
+namespace geosx {
+
+class ProjectionEDFMHelper {
+ public:
+  ProjectionEDFMHelper( MeshLevel const & mesh,
+                        GeometricObjectManager const * geometricObjManager,
+                        std::string const & coeffName,
+                        CellElementStencilTPFA & stencil );
+
+  // add Fracture-matrix connections to the cell stencil
+  void addNonNeighboringConnections(EmbeddedSurfaceSubRegion const & fractureSubRegion);
+
+  virtual ~ProjectionEDFMHelper() = default;
+
+ private:
+
+  /* This is a convenience struct to avoid passing too many argyuments (one object instead of three indices) */
+  struct CellID
+  {
+    CellID(localIndex r, localIndex sr, localIndex i)
+        : region(r), subRegion(sr), index(i)
+    {}
+
+    localIndex region;
+    localIndex subRegion;
+    localIndex index;
```

Comment:
We need more of there ;)

Reply:
Copy the triple index data into a struct for passing between functions called inside a kernel? Am I reading this correctly?

Reply:
@rrsettgast Is passing three contiguous values of elementary types much more expensive than passing three separate entities of these elementary types/

Reply:
Although I am passing it by reference, I still wonder.